python/GafferSceneUI/EditScopeUI.py

```diff
@@ -101,3 +108,58 @@ def __pruningKeyPress( viewer, event ) :
 			GafferScene.EditScopeAlgo.setPruned( editScope, selection, True )
 
 	return True
+
+def __visibilityKeyPress( viewer, event ) :
+
+	if not ( event.key == "H" and event.Modifiers.Control ) :
+		return False
+
+	if not isinstance( viewer.view(), GafferSceneUI.SceneView ) :
+		return False
+
+	editScope = viewer.view().editScope()
+	if editScope is None or Gaffer.MetadataAlgo.readOnly( editScope ) :
+		return False
+
+	viewedNode = viewer.view()["in"].getInput().node()
+	if editScope != viewedNode and editScope not in Gaffer.NodeAlgo.upstreamNodes( viewedNode ) :
+		# Spare folks from hiding things in a downstream EditScope.
+		## \todo When we have a nice Viewer notification system we
+		# should emit a warning here.
+		return True
```

Comment:
I think this is unnecessary - the inspector won't be editable if the EditScope isn't in the history.

Reply:
Right, inspectors will take care of that for us, fixed and squashed into https://github.com/GafferHQ/gaffer/pull/5381/commits/ed1b6a582be422aa6c6a45587fda3bd794a284ba